scripts/build-and-test.sh

```diff
@@ -1,80 +1,99 @@
 #!/usr/bin/env bash
-# We use set -e to bail on first non zero exit code of any processes launched
-# and -x to exit upon any unbound variable. -x will output command lines used
-# (with variable expansion)
-set -eux
 
-# brew install bash (4) to get this working on OSX!
+set -euxo pipefail
+# The above bash options do the following:
+
+# -e When this option is on, if a simple command fails for any of the reasons
+#    listed in Consequences of Shell Errors or returns an exit status value >0,
+#    and is not part of the compound list following a while, until, or if
+#    keyword, and is not a part of an AND or OR list, and is not a pipeline
+#    preceded by the ! reserved word, then the shell shall immediately exit.
+# -u The shell shall write a message to standard error when it tries to expand a
+#    variable that is not set and immediately exit. An interactive shell shall
+#    not exit.
+# -x The shell shall write to standard error a trace for each command after it
+#    expands the command and before it executes it. It is unspecified
+#    whether the command that turns tracing off is traced.
+# -o pipefail
+#    Pipelines fail on the first command which fails instead of dying later on
+#    down the pipeline.
+
 shopt -s globstar
 
 ################################## ENVIRONMENT #################################
 
 # If not CI, then set some defaults
-if [[ "${CI:-}" == "" ]]; then
-  TRAVIS_BRANCH=${TRAVIS_BRANCH:-feature}
-  CC=${CC:-gcc}
-  ADDRESS_MODEL=${ADDRESS_MODEL:-64}
-  VARIANT=${VARIANT:-debug}
+if [[ -z ${CI:-} ]]; then
+  : ${TRAVIS_BRANCH:=feature}
+  : ${CC:=gcc}
+  : ${ADDRESS_MODEL:-64}
```

Comment:
Hmm, so the effect is that when `build_and_run.sh` is run on a non-Travis environment, it will produce a 32-bit executable?